### ISSUE TITLE
Actionlint might be wrong about some syntax checks

### ISSUE THREAD

**Issue by AkechiShiro** [rhysd/actionlint#144]
Official documentation shows this example : https://github.com/actions/checkout#push-a-commit-using-the-built-in-token

```yml
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - run: |
          date > generated.txt
          git config user.name github-actions
          git config user.email github-actions@github.com
          git add .
          git commit -m "generated"
          git push
```
Which does combine "run" and "uses" sections but does not trigger the syntax warning, I'm having.

I'm having this syntax check error : step must run script with "run" section or run action with "uses" section
But I'm not sure, I understand exactly why, even if the reason is mentioned.

**Comment by rhysd** [comment]
I tried your example workflow in my local and actionlint didn't report any error.

**Comment by AkechiShiro** [comment]
I have an example that I could submit but it's a bit long, I just didn't understand really clearly why I could have run and uses in the same job, but actually we can have both no ? It's just that run needs to be in some kind of order perhaps ?

**Comment by rhysd** [comment]
'run' and 'uses' must be put in separate steps.

**Comment by AkechiShiro** [comment]
Alright then any idea why is this failing, uses and run are put in separate steps no ? 

It just fails with some yaml parsing issue, I believe but sometimes when I tweak it, it fails with 'run' and 'uses' cannot be used together on the same step.
```yaml
name: Build on MacOS
on: [push]
jobs:
  build-action:
    runs-on: macos-latest
    steps:
      - name: Building
      - uses: actions/checkout@v3
      - run: | 
          chmod +x ./build.sh
          ./build.sh
          mkdir artifacts && cp ./metacall-tarball-macos-x64.zip artifacts
  upload-tarball: 
    name: Upload build tarball production artifacts
    needs: build-action
    runs-on: macos-latest
    steps:
      - name: metacall-tarball
      - uses: actions/upload-artifact@v3
        with: 
          name: metacall-tarball
          path: ./artifacts/metacall-tarball-macos-x64.zip
  python-tests:
    name: Python loader tests
    needs: [build-action, upload-tarball]
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [macos-latest, macos-12]
    steps:
      - name: Download build/artifact from cache
      - uses: actions/download-artifact@v3
        with:
          name: metacall-tarball
          path: ./artifacts/metacall-tarball-macos-x64.zip
      - run: | 
          lots_of_commands
```
